megamek/src/megamek/client/ui/swing/phaseDisplay/FiringDisplay.java

```diff
@@ -1502,6 +1502,7 @@ protected void updateForNewAction() {
         }
         updateTarget();
         updateDonePanel();
+        refreshAll();
```

Comment:
So whenever an action that requires an update for new action, like torso twist or flip arms happen it will then call refresh all, who calls `update firing arc`, who calls `show firing solutions`, solving the problem.